broker-core/src/main/java/io/zeebe/broker/workflow/processor/CatchEventBehavior.java

```diff
@@ -134,7 +134,9 @@ public boolean occurEventForElement(
       workflowInstanceRecord.wrap(source);
       workflowInstanceRecord.setPayload(eventPayload);
       workflowInstanceRecord.setElementId(eventHandlerId);
-      workflowInstanceRecord.setBpmnElementType(BpmnElementType.EVENT);
+      if (!source.getElementId().equals(eventHandlerId)) {
+        workflowInstanceRecord.setBpmnElementType(BpmnElementType.BOUNDARY_EVENT);
```

Comment:
We should add at least a comment here in which case we need to change the element type, and when not. Otherwise, we could also lookup for the element and get the type of it.